examples/thermomechanics/thermal_deformation_heat_transfer.cpp

```diff
@@ -0,0 +1,162 @@
+                                             typename thermal_type::base_type>>(
+            exec_space(), low_corner, high_corner, num_cells, halo_width );
+
+    // ====================================================
+    //            Custom particle initialization
+    // ====================================================
+    auto rho = particles->sliceDensity();
+    auto temp = particles->sliceTemperature();
+    auto init_functor = KOKKOS_LAMBDA( const int pid )
+    {
+        // Density
+        rho( pid ) = rho0;
+        // Temperature
+        temp( pid ) = temp0;
+    };
+    particles->updateParticles( exec_space{}, init_functor );
+
+    // ====================================================
+    //                    Force model
+    // ====================================================
+    auto force_model = CabanaPD::createForceModel(
+        model_type{}, CabanaPD::Elastic{}, *particles, delta, K, kappa, cp,
+        alpha, temp0 );
+
+    // ====================================================
+    //                   Create solver
+    // ====================================================
+    auto cabana_pd = CabanaPD::createSolverElastic<memory_space>(
+        inputs, particles, force_model );
+
+    // ====================================================
+    //                   Boundary condition
+    // ====================================================
+    // Temperature profile imposed on top, bottom, left, and right surfaces
+    double dy = particles->dx[1];
+    using plane_type = CabanaPD::RegionBoundary<CabanaPD::RectangularPrism>;
+
+    // Top surface: y-direction
+    plane_type plane1( low_corner[0], high_corner[0], high_corner[1] - dy,
+                       high_corner[1] + dy, low_corner[2], high_corner[2] );
+
+    // Bottom surface: y-direction
+    plane_type plane2( low_corner[0], high_corner[0], low_corner[1] - dy,
+                       low_corner[1] + dy, low_corner[2], high_corner[2] );
+
+    std::vector<plane_type> planes = { plane1, plane2 };
+    temp = particles->sliceTemperature();
+    // This is purposely delayed until after solver init so that ghosted
+    // particles are correctly taken into account for lambda capture here.
+    auto temp_bc = KOKKOS_LAMBDA( const int pid, const double )
+    {
+        temp( pid ) = 0.0;
+    };
+    auto bc = CabanaPD::createBoundaryCondition(
+        temp_bc, exec_space{}, *particles, planes, false, 1.0 );
+
+    // ====================================================
+    //                   Simulation run
+    // ====================================================
+    cabana_pd->init( bc );
+    cabana_pd->run( bc );
+
+    // ====================================================
+    //                      Outputs
+    // ====================================================
+
+    // Output temperature along the y-axis
+    auto temp_output = particles->sliceTemperature();
```

Comment:
Do we need to use a different variable here (as opposed to "temp") or it this just for convenience/clarity?